### DISCUSSION TITLE
How regex rules work?

### DISCUSSION THREAD

**Discussion by Cerebellum-ITM** [FlorianWoelki/obsidian-icon-folder#538] [Q&A]
I've been creating rules for my folders and icons. Sometimes they work without problems other times it seems to ignore them all leaving the "." ruler icon. In some messages I have seen that they use something like "folder/*" the name can change and does it mean that it applies only to that directory? Could someone explain to me how to apply my rules correctly. Thanks for your comments

**Comment by iamzweiaa** [comment]
me too, same here
like i have the daily note named as YYYY-MM-DD
and in regex i put it as ^\d{4}-\d{2}-\d{2}$
but it doesn't work

**Comment by CGPGrey** [comment]
+1 

Really interested in the custom rules, but can't get them to work.

**Comment by CHerSun** [comment]
Would love a bit of clarification too. Trying to add icon to external links (starting with http(s) - `^https?:\/` is the pattern)

Update: tested with just `.` pattern - all file-related links were highlighted, but not external links to web-pages. Looks like the thing I wanted is not achievable

**Comment by HarleyStroud** [comment]
I'm also curious about this, anyone have an update? I just want all folders to have one icon, and all notes (.md) files to have another icon, and nothing has worked so far.

**Comment by vergelli** [comment]
Any update here?

**Comment by canemacchina** [comment]
look at https://github.com/FlorianWoelki/obsidian-iconize/blob/main/src/lib/custom-rule.ts#L36 and https://github.com/FlorianWoelki/obsidian-iconize/blob/main/src/lib/custom-rule.ts#L165

Basically, it first checks if the rule applies to the filetype (folder, file or both), then it checks if the rule regex matches the file name or the path, depending on whether `useFilePath` is true or false.

In the custom rule edit popup, you can set if the entire path should be used to match the regex, and choose if the rule must be applied to files, folders or both.

@HarleyStroud for your case, for example, you just need to create two rules, both using `.` as the regex, one that applies to just folders, one to just files.
